storage-node-v2/src/services/runtime/transactionNonceKeeper.ts

```diff
@@ -16,19 +26,36 @@ const lock = new AwaitLock()
  * @returns promise with transaction nonce for a given account.
  *
  */
-export async function getNonce(api: ApiPromise, account: KeyringPair): Promise<Index> {
+export async function getTransactionNonce(api: ApiPromise, account: KeyringPair): Promise<Index> {
   await lock.acquireAsync()
   try {
-    if (nonce === null) {
+    let nonce: Index | undefined = nonceCache.get(nonceEntryName)
+    if (nonce === undefined) {
       nonce = await api.rpc.system.accountNextIndex(account.address)
     } else {
       nonce = nonce.add(new BN(1)) as Index
     }
```

Comment:
It won't fundamentally solve the issue and the issue itself shouldn't arise during the intended usage because the storage provider as a worker should have a dedicated role account. My solution would be organizational - use some restrictions: "The storage node's worker account cannot be used to issue any custom transactions with running storage node.". The obvious exception would be the CLI commands and currently, the storage node must be stopped to run them. To mitigate this we could create a special API for commands via REST on the running storage provider in the future versions. Another option would be to introduce several accounts in the working group to perform operations. @mnaamani might probably review this conversation